app/src/main/java/com/iic/shopingo/ui/ContactDetailsActivity.java

```diff
@@ -44,26 +51,56 @@ protected void onCreate(Bundle savedInstanceState) {
 
   @OnClick(R.id.contact_save_btn)
   void onSaveClicked() {
-    updateUser();
-    Intent intent = new Intent(this, HomeActivity.class);
-    intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-    startActivity(intent);
-    finish();
+    if (validateInput()) {
```

Comment:
You can remove the nesting by checking if the input is invalid and returning from the method.